release.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+function main() {
+  parse_args $@
+  
+  if [ $h ]; then 
+      help; exit 0
+  fi
+  if ! [ $d ]; then 
+      check_dependencies
+  fi
+  if [ $t ]; then
+      if [ "$RELEASE_TAG" = "" ]; then
+          echo "Invalid release tag"; exit 1
+      fi
+  else 
+      echo "Release tag must be passed via the -t flag"
+  fi     
+  if [ $i ]; then
+      if ! [[ "$TICKET_NUMBER" =~ ^[0-9]+$ ]]; then
+          echo "Invalid ticket number"; exit 1
+      fi
+  else 
+      echo "Issue/ticker number must be passed via the -i flag"
+  fi 
+
+
+  if ! [ $f ]; then
+      start_release
+  else
+      if ! [ $k ]; then
+          echo "Signing key not provided"; exit 1
+      fi
+      if ! [[ "$SIGNING_KEY" =~ ^[a-zA-Z0-9]+$ ]]; then
+          echo "Invalid format for signing key"; exit 1
+      fi
+      complete_release
+  fi
+
+}
+
+function parse_args() {
+    local param
+    while getopts ":cdhi:t:fk:" param; do
+        case $param in
+            c) c=true;;
+            d) d=true;;
+            f) f=true;;
+            h) h=true;;
+            i) i=true; TICKET_NUMBER="$OPTARG";;
+            k) k=true; SIGNING_KEY="$OPTARG";;
+            t) t=true; RELEASE_TAG="$OPTARG";;
+            \?) help; exit 1;;
+        esac
+    done
+    shift $((OPTIND-1)) #clear argument list
+}
+
+
+function clean() {
+  rm -rf tahoe-release-*
+}
+
+function check_dependencies() {
+    [[ -z $(git status -s) ]] || { echo >&2 "repo is not clean, commit everything first!"; exit 1; }
+    python -c "import wheel" || { echo >&2 "wheel is not installed. Install via pip!"; exit 1; }
+}
+
+function start_release() {
+  # create release clone, and start release tasks    
+  git clone https://github.com/tahoe-lafs/tahoe-lafs.git "../tahoe-release-$RELEASE_TAG"
+  cd "../tahoe-release-$RELEASE_TAG"
+  python -m venv venv
+  ./venv/bin/pip install --editable .[test]
+  git branch "$TICKET_NUMBER.release-$RELEASE_TAG" # looks like XXXX.release-1.16.0
+  git checkout "$TICKET_NUMBER.release-$RELEASE_TAG"
+  ./venv/bin/tox -e news
+  touch "newsfragments/$TICKET_NUMBER.minor"
+  RELEASE_TITLE="Release $RELEASE_TAG $(date +('%Y-%m-%d'))"
+  git add . && git commit -s -m "tahoe-lafs-$RELEASE_TAG news"
+  sed -i -r 's/(\.){2}[[:space:]]towncrier start line//g' NEWS.rst 
```

Comment:
POSIX standard doesn't specify -r or -E but it seems rather widespread so it might be fine for purposes of release automation. There probably isn't anything here that couldn't be written as BRE but it's messy as it is.
If this is deleting whole line you can use the d command instead of s. That is `/pattern/d` instead of `s/pattern//`